### ISSUE TITLE
Failing to load some images

### ISSUE THREAD

**Issue by ghost** [square/picasso#742]
I'm not being to load some specific images from the server. I'm using picasso-2.3.4. I get an error with or without okhttp.

This image doesn't load:
http://az637658.vo.msecnd.net/public-data/d7d94e4c-6c3d-4c24-97d2-4eabdcf235d8.jpg

This one does:
http://az637658.vo.msecnd.net/public-data/2129df6f-3056-4464-b4ea-126c094e4ffe.jpg

My loading code:

```
        Picasso picasso = new Picasso.Builder(this)
                .listener(new Picasso.Listener() {
                    @Override
                    public void onImageLoadFailed(Picasso picasso, Uri uri, Exception e) {
                        e.printStackTrace();
                    }
                }).loggingEnabled(true).build();

        picasso.load(mPhoto.getPhotoUrl())
                .fit().centerCrop()
                .into(viewHolder.imageViewPhotoDetail);
```

The error (with okhttp and okio):
com.squareup.picasso.Downloader$ResponseException: 504 Unsatisfiable Request (only-if-cached)

The error (without okhttp and okio):
 com.squareup.picasso.Downloader$ResponseException: 502 Bad Gateway

Regarding the image HTTP headers, I haven't seen any major differences between them.

**Comment by valtinho** [comment]
Where can I download a picasso jar that includes this "MarkableInputStream in NetworkRequestHandler" fix? I have been using the latest version (2.5.2) and this bug is still there. I am getting the 502 bad gateway exactly as described by dwbrito. In addition, I am seeing that this is a known bug that has been around since at least version 2.0.2. I downloaded the source code, but when I try to generate a jar out of it it does not work. Please help.

**Comment by flipnovidade** [comment]
Addition of the following 2 attributes in (AndroidManifest.xml) worked for me:

android:largeHeap="true"
android:hardwareAccelerated="false"
